### PR TITLE
PhpUnitTestAnnotationFixer: Do not prepend with test if method is test()

### LINKED ISSUE #3437: PhpUnitTestAnnotationFixer false positive since 2.10
It now duplicates `test()` name to `testTest()`. It didn't happen before 2.10.

What was changed?

image

See also Travis: https://travis-ci.org/Symplify/Symplify/jobs/327849604#L644

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] adds a failing test case
* [x] fixes the edge case

Fixes #3437.

### CODE REVIEW

File: src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php [hunk 1]
```diff
@@ -296,10 +296,14 @@ private function getDocBlockIndex(Tokens $tokens, $index)
      */
     private function hasTestPrefix($functionName)
     {
-        if (!$this->startsWith('test', $functionName) || strlen($functionName) < 5) {
+        if (!$this->startsWith('test', $functionName)) {
             return false;
         }
 
+        if (4 === strlen($functionName)) {
```
Comment (keradus) [inline src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php]: what we do for <4 now ?

Comment (SpacePossum) [reply src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php]: `abc` becomes `testAbc` IIUC

Comment (localheinz) [reply src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php]: We won’t get this far if `$functionName` doesn’t start with `test`, see condition above.

File: src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php [hunk 2]
```diff
@@ -296,10 +296,14 @@ private function getDocBlockIndex(Tokens $tokens, $index)
      */
     private function hasTestPrefix($functionName)
     {
-        if (!$this->startsWith('test', $functionName) || strlen($functionName) < 5) {
+        if (!$this->startsWith('test', $functionName)) {
             return false;
         }
 
+        if ('test' === $functionName) {
```
Comment (localheinz) [inline src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php]: @keradus @SpacePossum 

Is this better?

Comment (SpacePossum) [reply src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php]: :+1: it is more explicit and faster, does this solution work for `TesT` etc. as well?

Comment (localheinz) [reply src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php]: I can adjust!

I think, though, that for PHPUnit it needs to be `test` (case-sensitive) to qualify as test.

Comment (localheinz) [reply src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php]: Also see [`PhpUnitTestAnnotationFixer::startsWith()`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.9.1/src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php#L257-L262), where we compare case-sensitively.

Comment (SpacePossum) [reply src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php]: ah that makes sense! never mind me :)

**Comment by keradus** [comment]
Thank you @localheinz.

**Comment by localheinz** [comment]
Thank you, @keradus and @SpacePossum!
